### ISSUE TITLE
bug: dist/esm/polyfills is empty 

### ISSUE THREAD

**Issue by AnnaHoege** [ionic-team/stencil#4661]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

v4.0.4

### Current Behavior

When running `npm run build`, the resulting dist/esm/polyfills folder is empty every other time the command is run. This causes issues as the loader exports the index file of the polyfills folder.

### Expected Behavior

The polyfills folder should contain at least an index file on every build.

### System Info

```shell
System: node 18.17.0
      Platform: windows (10.0.19045)
      CPU Model: Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz (8 cpus)
      Compiler: C:\Users\AHOEGE\Projects\stencil-bug-reproduction\bug-reproduction\node_modules\@stencil\core\compiler\stencil.js
      Build: 1691405536
      Stencil: 4.0.4
      TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.19.2
```

### Steps to Reproduce

1. Create a new project via `npm init stencil`
2. Run `npm i`
3. Run `npm build`
4. Run `npm build`

### Code Reproduction URL

https://github.com/AnnaHoege/stencil-bug-reproduction

**Comment by FirstVertex** [comment]
See this intermittently on my Windows 11 machine.  `stencil build` completes and the dist folder is generated, there's a polyfills folder under esm/ but it is empty.  When I run build again it is fixed.

**Comment by FirstVertex** [comment]
Please, make it stop.  All week long I have to build twice.  1 minute per build.  Building dozens of times per day as I figure out our Storybook integration.  So much wasted time.  Please, this bug needs a high priority and severity.

**Comment by rwaskiewicz** [comment]
Can anyone running into this issue please try out this dev build for me and let me know if this works?
```
npm i @stencil/core@4.4.1-dev.1697228154.e0f8280
```
Thanks!

**Comment by imagoiq** [comment]
Hi,

Works on my side, the polyfills folder is not empty anymore and contains: `core-js.js`, `dom.js`, `es5-html-element.js`, `index.js` and `system.js`. 

Thanks for handling it.

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.6.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.6.0)
